src/sys/unix/rw_lock.rs

```diff
@@ -67,4 +67,23 @@ impl<T: AsRawFd> RwLock<T> {
         // use `AsFd` instead of `AsRawFd` and eliminate this `unsafe` block.
         unsafe { BorrowedFd::borrow_raw(self.inner.as_raw_fd()) }
     }
+
+    pub(crate) fn acquire_read_lock(&self) -> io::Result<()> {
```

Comment:
Following the same approach I think we can just put a bunch of `acuire_*` method in platform `rw_lock`, calling them from the platform-independent `rw_lock` and remove all platform-specific `read/write_guard`